perf/lua/luafun.lua

```diff
@@ -0,0 +1,77 @@
+local fun = require("fun")
+local clock = require("clock")
+local t = require("tarantool")
+local benchmark = require("benchmark")
+
+local USAGE = [[
+ This benchmark measures the performance of luafun module.
+]]
+
+local params = benchmark.argparse(arg, {}, USAGE)
+
+local bench = benchmark.new(params)
+
+local _, _, build_type = string.match(t.build.target, "^(.+)-(.+)-(.+)$")
+if build_type == "Debug" then
+    print("WARNING: tarantool has built with enabled debug mode")
+end
+
+--
+-- Data for `drop_while` benchmark.
+--
+
+local DROP_WHILE_CYCLES = 10^3
+local DROP_WHILE_N = 10^7
+
+local function DROP_WHILE_FILTER(x)
+    return x <= 50
+end
+
+local DROP_WHILE_DATA = {}
+for _ = 1, DROP_WHILE_N / 2 do
+    local value = math.random(50)
+    table.insert(DROP_WHILE_DATA, value)
+end
+for _ = 1, DROP_WHILE_N / 2 do
+    local value = math.random(51, 100)
+    table.insert(DROP_WHILE_DATA, value)
+end
+
+--
+-- Benchmarks themselves.
+--
+
+local tests = {{
+    name = "drop_while",
+    items = DROP_WHILE_N,
+    cycles = DROP_WHILE_CYCLES,
+    payload = function()
+        local acc = 0
+        for _ = 1, DROP_WHILE_CYCLES do
+            -- Random factor to disable any possible optimizations
+            local factor = math.random(10)
+            for _, i in fun.iter(t):drop_while(DROP_WHILE_FILTER) do
```

Comment:
When I ran this benchmark locally, I got the following results (see also [the results in our Grafana](https://mon.tarantool.io/grafana/d/olYWJGsSk/performance-alpha-perf_micro?orgId=1&from=now-24h&to=now&var-bucket=perf-debug&var-measurement=luafun&var-branch=drewdzzz%2Fpairs_drop_while&var-subtest=All)).
```sh
LUA_PATH="perf/lua/?.lua;;" src/tarantool -jdump=i perf/lua/luafun.lua --output_format=console

12143323682686 rps
```
I believe that LuaFun is fast, but this is too fast :).

I suppose you mean `DROP_WHILE_DATA` instead of `t` here.

Reply:
> 12143323682686 rps

Blazingly fast, like it's written in Rust 😄 
Thanks for noticing! Fixed!